site/src/request_handlers/self_profile.rs

```diff
@@ -638,67 +586,38 @@ pub async fn handle_self_profile(
         .set(
             Tag::Scenario,
             selector::Selector::One(body.scenario.clone()),
+        )
+        .set(
+            Tag::Metric,
+            selector::Selector::One("cpu-clock".to_string()),
         );
 
-    let mut commits = vec![index
-        .commits()
-        .into_iter()
-        .find(|c| c.sha == *body.commit.as_str())
-        .map(|c| ArtifactId::Commit(c))
-        .or_else(|| {
-            index
-                .artifacts()
-                .find(|a| **a == body.commit)
-                .map(|a| ArtifactId::Tag(a.to_owned()))
-        })
-        .ok_or(format!("could not find artifact {}", body.commit))?];
+    // Helper for finding an `ArtifactId` based on a commit sha
+    let find_aid = |commit: &str| {
+        index
+            .commits()
+            .into_iter()
+            .find(|c| c.sha == *commit)
+            .map(|c| ArtifactId::Commit(c))
+            .or_else(|| {
+                index
+                    .artifacts()
+                    .find(|a| *a == commit)
+                    .map(|a| ArtifactId::Tag(a.to_owned()))
+            })
+            .ok_or(format!("could not find artifact {}", body.commit))
```

Comment:
Probably this should be `commit`, not `body.commit`?

It may make sense to move this to a function on Index, FWIW.